pages/templates/_slug.vue

```diff
@@ -67,7 +67,10 @@ export default {
     MoreTemplates,
     Faq,
   },
-  async asyncData({ $axios, params }) {
+  async asyncData({ $axios, params, payload }) {
+    if (payload) {
```

Comment:
what is payload?

Reply:
payload contains both templates and categories.. and this is use while generation.. In the generate method in nuxt config we pass these payload to these routes..

Reply:
They why do we need to make API call again?

Reply:
So generate will work only while generating the site and it will generate a static site for production..
But during development (in local) it doesn't use the generate method..

Reply:
In Nuxt 2, when you run the nuxt generate command, it pre-renders your static site and creates payloads for each route to optimize loading times. These payloads contain the data fetched during the generation process and are sent to the routes so that when a user visits a pre-rendered page, the data is already there, making the page load faster.

However, during development (nuxt dev), the site is not statically generated. Instead, it runs as a server-rendered application where each page is dynamically rendered on request. This means that payloads are not created or sent during development because the pages are rendered on the fly with fresh data fetched from your APIs or other data sources.

This difference in behavior is by design, as development mode is meant to give you the most up-to-date view of your site, whereas the generate command is for creating a static site optimized for production.

Reply:
If that's the case than. We should add a comment or readme to use generate. Having 2 methods will confuse and while upadting the code if dev only updates the code used for development and forgot production one then it'll break.

Reply:
I think we can use the same utils method here as well.. So if we do so, then we will have the logic at one place.

Reply:
There could be other ways too. To handle prod and dev with the same code. Having 2 different code/logics will be hard to maintain

Reply:
I have updated the util. Now the logic is in one place.